crates/red_knot_python_semantic/resources/mdtest/annotations/annotated.md

```diff
@@ -0,0 +1,79 @@
+# `Annotated`
+
+`Annotated` attaches arbitrary metadata to a given type.
+
+## Usages
+
+`Annotated[T, ...]` is equivalent to `T`: All metadata arguments are simply ignored.
+
+```py
+from typing_extensions import Annotated
+
+def _(x: Annotated[int, "foo"]):
+    reveal_type(x)  # revealed: int
+
+def _(x: Annotated[int, lambda: 0 + 1 * 2 // 3, _(4)]):
+    reveal_type(x)  # revealed: int
+
+def _(x: Annotated[int, "arbitrary", "metadata", "elements", "are", "fine"]):
+    reveal_type(x)  # revealed: int
+
+def _(x: Annotated[tuple[str, int], bytes]):
+    reveal_type(x)  # revealed: tuple[str, int]
+```
+
+## Parametrization
+
+It is invalid to parametrize `Annotated` with less than two arguments.
+
+```py
+from typing_extensions import Annotated
+
+# TODO: This should be an error
+def _(x: Annotated):
+    reveal_type(x)  # revealed: Unknown
+
+# error: [invalid-annotated-parameter]
+def _(x: Annotated[()]):
+    reveal_type(x)  # revealed: Unknown
+
+# error: [invalid-annotated-parameter]
+def _(x: Annotated[(int,)]):
+    reveal_type(x)  # revealed: int
+
+# `Annotated[T]` is invalid and will raise an error at runtime,
+# but we treat it the same as `T` to provide better diagnostics later on.
+# The subscription itself is still reported, regardless.
+# error: [invalid-annotated-parameter]
+def _(x: Annotated[int]):
```

Comment:
it's easy to miss that there's an assertion here when reading this; it gets a little lost in the multiline descriptive comment:

```suggestion
def _(x: Annotated[int]):  # error: [invalid-annotated-parameter]
```